doc/build.md

```diff
@@ -4,36 +4,95 @@
 
-This document guides you through building Nanvix. You can either use native tools or a Docker image.
+This document guides you through building Nanvix. You can either use the `z` utility script for a
+simplified build process or do it manually.
 
 ## Table of Contents
 
-- [Building Nanvix with Docker](#building-nanvix-with-docker)
-- [Building Nanvix with Native Tools](#building-nanvix-with-native-tools)
-- [Default Build Parameters](#default-build-parameters)
-- [List of Optional Build Parameters](#list-of-optional-build-parameters)
+- [Building Nanvix with `z` (Preferred Method)](#building-nanvix-with-z-preferred-method)
+  - [Getting Started with `z`](#getting-started-with-z)
+  - [Using `z` to Build Nanvix with Docker](#using-z-to-build-nanvix-with-docker)
+  - [Using `z` to Build Nanvix with a Local Toolchain](#using-z-to-build-nanvix-with-a-local-toolchain)
+- [Building Nanvix Manually](#building-nanvix-manually)
+  - [Manually Building Nanvix with Docker](#manually-building-nanvix-with-docker)
+  - [Manually Building Nanvix with a Local Toolchain](#manually-building-nanvix-with-a-local-toolchain)
+- [Build Parameters](#build-parameters)
+  - [Default Build Parameters](#default-build-parameters)
+  - [Optional Build Parameters](#optional-build-parameters)
 
-## Building Nanvix with Docker
+## Building Nanvix with `z` (Preferred Method)
 
-> ℹ️ This builds Nanvix with the default parameters except for the toolchain directory.
+`z` is a utility for building Nanvix. It provides you with a simplified interface for building
+Nanvix either using Docker or your local toolchain.
+
+### Getting Started with `z`
+
+For more information on how to use the `z` utility, you can run:
+
+```bash
+./z help
+```
+
+### Using `z` to Build Nanvix with Docker
+
+To build Nanvix using the latest Docker image and default build parameters, simply run:
+
+```bash
+./z build --with-docker -- all
+```
+
+> ℹ️  Refer to the [Build Parameters](#build-parameters) section for more information on how to
+customize the build process.
+
+### Using `z` to Build Nanvix with a Local Toolchain
+
+To build Nanvix using your local toolchain and default build parameters, simply run:
+
+```bash
+./z build -- all
+```
+
+> ℹ️  Refer to the [Build Parameters](#build-parameters) section for more information on how to
+customize the build process.
+
+## Building Nanvix Manually
+
+Instead of using the `z` utility, you can build Nanvix manually.
+
+### Manually Building Nanvix with Docker
+
+To build Nanvix using the latest Docker image and default build parameters, simply run:
 
 ```bash
 docker run \
   -it --rm -v"$(pwd):/mnt" \
   nanvix/toolchain \
-  /bin/bash -l -c \
-  "cd /mnt ; git config --global --add safe.directory '*' ; make TOOLCHAIN_DIR=/opt all ; chown -R 1000:1000 ."
+  /bin/bash -l -c "\
+    cd /mnt ; \
+    git config --global --add safe.directory '*' ; \
+    make TOOLCHAIN_DIR=/opt all ; \
```

Comment:
The toolchain dir here looks different than the one in the `./z` command. Is this expected?